esrally/metrics.py

```diff
@@ -114,6 +114,27 @@ def guarded(self, target, *args, **kwargs):
 
             try:
                 return target(*args, **kwargs)
+            except elasticsearch.helpers.BulkIndexError as e:
+                for err in e.errors:
+                    err_type = err.get("index", {}).get("error", {}).get("type", None)
+                    if err.get("index", {}).get("status", None) not in (502, 503, 504, 429):
```

Comment:
Sure, and I'm not sure I understand your last two sentences:
> I know you did not come up with the list, but I'm not sure I find the list very convincing. Even 429 seems OK to retry after having slept.

We _do_ retry on 429s, but I can see that it can be easily misread. A side effect of using a constant is that it's now much more 'readable' (IMHO).

Addressed in https://github.com/elastic/rally/pull/1703/commits/1d88020e21109b6afd1ddb2295e4c05ea8dd78fd